osf/utils/workflows.py

```diff
@@ -165,6 +165,26 @@ def from_transition(cls, from_state, to_state):
         return transition_to_trigger_mappings.get((from_state, to_state))
 
 
+class CollectionSubmissionsTriggers(ModerationEnum):
+    '''The acceptable 'triggers' to use with a CollectionSubmissionsAction'''
+    SUBMIT = 0
+    APPROVE = 1
+    REJECT = 2
+    ADMIN_REMOVE = 3
+    MODERATOR_REMOVE = 4
+
+    @classmethod
+    def from_transition(cls, from_state, to_state):
+        transition_to_trigger_mappings = {
+            (ApprovalStates.IN_PROGRESS, ApprovalStates.UNAPPROVED): cls.SUBMIT,
```

Comment:
These transitions can wait until state machine implementation if you want, but should look like:

UNAPPROVED -> PENDING_MODERATION = SUBMIT
UNAPPROVED -> APPROVED = SUBMIT (unmoderated)
PENDING_MODERATION -> MODERATOR_REJECTED = REJECT
PENDING_MODERATION -> APPROVED = ACCEPT
APPROVED -> REMOVED = REMOVE (REMOVED needs to be added to ApprovalStates)
MODERATOR_REJECTED -> PENDING_MODERATION = RESUBMIT
REMOVED -> PENDING_MODERATION = RESUBMIT
REMOVED ->  APPROVED = RESUBMIT (unmoderated)

Reply:
See state machine in analysis doc: https://docs.google.com/document/d/1E2bvkurIrPTWv3rOmDSKIr6ggszuhbh6LuqNeTyLVkk/edit#